core/controllers/email_dashboard.py

```diff
@@ -215,3 +216,28 @@ def post(self, query_id):
         query_model.query_status = feconf.USER_QUERY_STATUS_ARCHIVED
         query_model.put()
         self.render_json({})
+
+
+class EmailDashboardTestBulkEmailHandler(base.BaseHandler):
+    """Handler for testing bulk email before sending it.
+
+    This handler sends a test email to submitter of query before it is sent to
+    qualfied users in bulk.
+    """
+    @require_valid_sender
+    def post(self, query_id):
+        query_model = user_models.UserQueryModel.get(query_id, strict=False)
+        if (query_model is None or
+                query_model.query_status != feconf.USER_QUERY_STATUS_COMPLETED):
+            raise self.InvalidInputException('400 Invalid query id.')
+
+        if query_model.submitter_id != self.user_id:
+            raise self.UnauthorizedUserException(
+                '%s is not an authorized user for this query.', self.user_id)
+
+        data = self.payload['data']
```

Comment:
Generally, for POST handlers, we do self.payload.get('email_subject'), etc. Not sure where the need for the extra 'data' key comes from...